src/chainstate/stacks/boot/costs.clar

```diff
@@ -0,0 +1,495 @@
+
+(define-read-only (cost_tuple_cons (n uint))
+    (runtime (nlogn n u1 u1)))
+
+(define-read-only (cost_add (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_sub (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_mul (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_div (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_geq (n uint))
+    (runtime u1))
+
+(define-read-only (cost_leq (n uint))
+    (runtime u1))
+
+(define-read-only (cost_le (n uint))
+    (runtime u1))
+
+(define-read-only (cost_ge  (n uint))
+    (runtime u1))
+
+(define-read-only (cost_int_cast (n uint))
+    (runtime u1))
+
+(define-read-only (cost_mod (n uint))
+    (runtime u1))
+
+(define-read-only (cost_pow (n uint))
+    (runtime u1))
+
+(define-read-only (cost_sqrti (n uint))
+    (runtime u1))
+
+(define-read-only (cost_xor (n uint))
+    (runtime u1))
+
+(define-read-only (cost_not (n uint))
+    (runtime u1))
+
+(define-read-only (cost_eq (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_begin (n uint))
+    (runtime u1))
+
+(define-read-only (cost_hash160 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_sha256 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_sha512 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_sha512t256 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_keccak256 (n uint))
+    (runtime u1))
+
+(define-read-only (cost_secp256k1recover (n uint))
+    (runtime u1))
+
+(define-read-only (cost_secp256k1verify (n uint))
+    (runtime u1))
+
+(define-read-only (cost_print (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_some_cons (n uint))
+    (runtime u1))
+
+(define-read-only (cost_ok_cons (n uint))
+    (runtime u1))
+
+(define-read-only (cost_err_cons (n uint))
+    (runtime u1))
+
+(define-read-only (cost_default_to (n uint))
+    (runtime u1))
+
+(define-read-only (cost_unwrap_ret (n uint))
+    (runtime u1))
+
+(define-read-only (cost_unwrap_err_or_ret (n uint))
+    (runtime u1))
+
+(define-read-only (cost_is_okay (n uint))
+    (runtime u1))
+
+(define-read-only (cost_is_none (n uint))
+    (runtime u1))
+
+(define-read-only (cost_is_err (n uint))
+    (runtime u1))
+
+(define-read-only (cost_is_some (n uint))
+    (runtime u1))
+
+(define-read-only (cost_unwrap (n uint))
+    (runtime u1))
+
+(define-read-only (cost_unwrap_err (n uint))
+    (runtime u1))
+
+(define-read-only (cost_try_ret (n uint))
+    (runtime u1))
+
+(define-read-only (cost_match (n uint))
+    (runtime u1))
+
+(define-read-only (cost_or (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_and (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_append (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_concat (n uint))
+    (runtime (linear n u1 u1)))
+
+(define-read-only (cost_as_max_len (n uint))
+    (runtime u1))
+
+(define-read-only (cost_contract_call (n uint))
```

Comment:
A `contract-call?` also has a storage read cost.

Reply:
Yep -- added this to the issue: https://github.com/blockstack/stacks-blockchain/issues/2058